### ISSUE TITLE
How does ESMC implement encoding proteins in a local computer, that is, converting sequences into arrays?

### ISSUE THREAD

**Issue by ht-234** [evolutionaryscale/esm#190]
Dear all, is there any tutorial that can be provided, thanks a lot. According to the official documentation, I try to use the following code but it always gives me an error saying: in _tokenize assert pad is not None

`from esm.models.esmc import ESMC
from esm.sdk.api import ESMProtein, LogitsConfig

protein = ESMProtein(sequence="AAAAA")
client = ESMC.from_pretrained("esmc_300m").to("cpu") # or "cpu"
protein_tensor = client.encode(protein)
logits_output = client.logits(
   protein_tensor, LogitsConfig(sequence=True, return_embeddings=True)
)
print(logits_output.logits, logits_output.embeddings)`

**Comment by ht-234** [comment]
I finally figured out why, the environment wasn't set up right, and following the instructions on hugging face to download tags.
  - biology
  - esm
  - protein
Special thanks to all of you who read my question, thanks!

**Comment by dhuvik** [comment]
Hi could you elaborate on what you did to get padding to work? I tried adding '<pad>' directly into the sequence s.t. it got 

``from esm.models.esmc import ESMC
from esm.sdk.api import ESMProtein, LogitsConfig

protein = ESMProtein(sequence="AAAA<pad><pad><pad>")
client = ESMC.from_pretrained("esmc_300m").to("cpu") # or "cpu"
protein_tensor = client.encode(protein)
logits_output = client.logits(
protein_tensor, LogitsConfig(sequence=True, return_embeddings=True)
)
print(logits_output.logits, logits_output.embeddings)``

and the size of the embeddings was [1,6,960] instead of [1,9,960]. Any advice on how you got around this would be appreciated.

**Comment by xtzhang-AI4S** [comment]
Hello, excuse me, how did you implement ESMC in a local computer? Can the parameters be downloaded locally?

**Comment by lzygitk7** [comment]
> Hello, excuse me, how did you implement ESMC in a local computer? Can the parameters be downloaded locally?

Yes. You can check this [page](https://huggingface.co/EvolutionaryScale).

**Comment by ht-234** [comment]
> 您好，您是如何在本地计算机中实施 ESMC 的？参数可以在本地下载吗？

Yes. The key to solving this problem is to configure the environment and download the various packages needed.
